tests/garage/tf/policies/test_categorical_mlp_policy_with_model.py

```diff
@@ -0,0 +1,77 @@
+import pickle
+from unittest import mock
+
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.envs import TfEnv
+from garage.tf.models import Model
+from garage.tf.policies import CategoricalMLPPolicyWithModel
+from tests.fixtures import TfGraphTestCase
+from tests.fixtures.envs.dummy import DummyDiscreteEnv
+
+
+class SimpleMLPModel(Model):
+    """Simple MLPModel for testing."""
+
+    def __init__(self, name, output_dim, *args, **kwargs):
+        super().__init__(name)
+        self.output_dim = output_dim
+
+    def _build(self, obs_input):
+        return tf.constant(
+            1., shape=(
+                1,
+                self.output_dim,
+            ))
+
+
+class TestCategoricalMLPPolicyWithModel(TfGraphTestCase):
+    def setUp(self):
+        super().setUp()
+        self.env = TfEnv(DummyDiscreteEnv())
+        self.env.reset()
+        self.obs, _, _, _ = self.env.step(1)
+        model_path = 'garage.tf.policies.' + \
+                     'categorical_mlp_policy_with_model.MLPModel'
+        with mock.patch(model_path, new=SimpleMLPModel):
+            self.policy = CategoricalMLPPolicyWithModel(env_spec=self.env.spec)
+
+    @mock.patch('numpy.random.rand')
+    def test_get_action(self, mock_rand):
+        mock_rand.return_value = 1
+        action, _ = self.policy.get_action(self.obs)
+        assert self.env.action_space.contains(action)
+        assert action == 0
+
+        actions, _ = self.policy.get_actions([self.obs])
+        for action in actions:
```

Comment:
does this loop run for more than one iteration?

Reply:
Yes if more than one observation is passed to `policy.get_actions`. I will add test to cover those cases.